.github/workflows/integrate-and-deploy.yml

```diff
@@ -54,12 +54,18 @@ jobs:
           TAG_LFMERGE=${TAG_LFMERGE:-latest}
           echo "TAG_LFMERGE=${TAG_LFMERGE}" >> $GITHUB_OUTPUT
       -
-        uses: actions/setup-node@v3
+        uses: pnpm/action-setup@v4
+      -
+        uses: actions/setup-node@v4
         with:
-          node-version: '16.14.0'
-          cache: 'npm'
+          node-version: '22.2.0'
+          cache: 'pnpm'
+      -
+        run: pnpm install
       -
-        run: npm ci
+        name: Prep for production build of legacy app
+        # docker compose build doesn't allow passing a `--target` parameter, so we have to replace the target in docker-compose.yml
+        run: "sed -i 's/target: development/target: production/g' docker-compose.yml"
```

Comment:
+1 for the compose merge solution.  That could also be a separate PR?

Reply:
I'll make it a separate PR to minimize the chances of merge conflicts (e.g., #1820 also touches workflows and I expect one merge conflict there, which will be easily resolved but if I'm making larger changes to the build process I might get harder-to-handle conflicts).

Reply:
Resolving conversation for now, just to unlock the merge button. Then I'll come back and unresolve it so that the suggestion is easy for me to find later.